### ISSUE TITLE
Z2-Converter : GetSmartContractState api takes very long time and blocks execution

### ISSUE THREAD

**Issue by chetan-zilliqa** [Zilliqa/zq2#1769]
Running the `GetSmartContractState` API takes approximately 4 minutes to return the output. During this processing period, the node is occupied with handling the request, and no other processing occurs on the node.

**Comment by maxconway** [comment]
@chetan-zilliqa, This has been changed around quite a bit since then I think, do we know if this is still the case? And can you tell me a bit more about the invocation that's causing this? Are we talking e.g. with very large `trie`s?

I've just run through the `get_smart_contract_sub_state` test case which exercises this and it looks like it's taking in the region of a second or two - too slow, but not 4 minutes

**Comment by maxconway** [comment]
Chucking a bit of timing code at the test case in release mode, I get the following:

Time to acquire lock: 1.803µs
Time to get latest block: 210.225µs
Time to get state: 461ns
Time to check if account exists: 7.745µs
Time to get account: 8.105µs
Time to get account trie: 7.975µs
Time to iterate over trie: 9.638µs
Total time: 277.692µs

So by far the slowest bit in this case is getting the block, so we'll need to make the test case more realistic if we want to work out what's going on

**Comment by chetan-zilliqa** [comment]
@maxconway , Yes ,this curl running on my standalone api node takes around 584 seconds on ZQ2 while it takes 26 seconds on ZQ1.
```
curl --request POST \
  --url http://34.87.132.124:4202/ \
  --header 'Content-Type: application/json' \
  --data '{
    "id": "1",
    "jsonrpc": "2.0",
    "method": "GetSmartContractState",
    "params": ["zil15lr86jwg937urdeayvtypvhy6pnp6d7p8n5z09"]
}'
```
![image](https://github.com/user-attachments/assets/3a10e007-a150-4a26-90d7-2f326de4c025)

**Comment by chetan-zilliqa** [comment]
![image](https://github.com/user-attachments/assets/6ac7220e-7f49-4400-90a1-619a9fff2956)


As we can see that ZQ2 time is higher but very marginally i say, But the panic regarding GetSmartContractState taking long time is fixed.

**Comment by maxconway** [comment]
Yeah I'd say it's still way too slow, but it doesn't need addressing separately to overall API performance enhancements
